### ISSUE TITLE
Get span duration while it is running (or/and start time)

### ISSUE THREAD

**Issue by ovidiubuligan** [open-telemetry/opentelemetry-cpp#2066]
Already exists for example in `C#` https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.duration?view=net-7.0 

**Is your feature request related to a problem?**
My problem is that I want to check from a different thread the duration of the currently executing span and alert if certain threshold is passed . For this I can't use the start time of the span since It is not available from the OTEL c++ api.

**Describe the solution you'd like**
New api for Span with `duration()` and `start_time()` 

Any thoughts?

**Comment by lalitb** [comment]
> New api for Span with duration() and start_time()

The otel specs doesn't requires any such API to be defined - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span-operations.  It would be interesting to know if any other language is implementing it. Probably .NET API is more generic, and needn't be fully otel compliant.

**Comment by lalitb** [comment]
@ovidiubuligan Do you have further question, and if this is good to close the issue.

**Comment by esigo** [comment]
please feel free to reopen if you have further question.

**Comment by ovidiubuligan** [comment]
I forgot to answer how I managed to do this without touching the business code. 
Basically I created a separate SpanProcessor that tracks span with start time in a _data structure_ with a WatchDog thread that checks the active spans duration and logs something if some span is taking too long. 

Might not be the optimal way since it involves extra processing but I didn't find a way of doing it without touching the business code. This way this WatchDogSpanProcessor can be reused .
